components/site/src/lib.rs

```diff
@@ -62,9 +62,10 @@ impl<'a> SerializedTaxonomyItem<'a> {
 impl Site {
     /// Parse a site at the given path. Defaults to the current dir
     /// Passing in a path is used in tests and when --root argument is passed
-    pub fn new<P: AsRef<Path>>(path: P, config_file: &str) -> Result<Site> {
+    pub fn new<P: AsRef<Path>, P2: AsRef<Path>>(path: P, config_file: P2) -> Result<Site> {
```

Comment:
It would force `path` and `config_file` to have the same type, what isn't necessarily required here. One may want to use for example &PathBuf and &str

But yes, currently this function is always called with `&PathBuf` so it doesn't matter that much + makes this line quite long.